File: data/multilang_comparison/program_go/src/func9.c
#include "program_go.h" 
array_t* func9(array_t_param* vars, const unsigned long PATH0, int loopsFactor) {
   size_t pCounter = vars->size;
   array_t* array207;
   if (pCounter > 0) {
      array207 = vars->data[--pCounter];
      array207->refC++;
      DEBUG_COPY(array207->id);
   } else {
      array207 = (array_t*)malloc(sizeof(array_t));
      array207->size = 176;
      array207->refC = 1;
      array207->id = 207;
      array207->data = (unsigned int*)malloc(array207->size*sizeof(unsigned int));
      memset(array207->data, 0, array207->size*sizeof(unsigned int));
      DEBUG_NEW(array207->id);
   }
   array_t_param params0;
   params0.size = 1;
   params0.data = (array_t**)malloc(params0.size*sizeof(array_t*));
   params0.data[0] = array207;
   array_t* array208 = func15(&params0, loopsFactor);
   DEBUG_RETURN(array208->id);
   free(params0.data);
   if(PATH0 & 1) {
      array_t* array211;
      if (pCounter > 0) {
         array211 = vars->data[--pCounter];
         array211->refC++;
         DEBUG_COPY(array211->id);
      } else {
         array211 = (array_t*)malloc(sizeof(array_t));
         array211->size = 818;
         array211->refC = 1;
         array211->id = 211;
         array211->data = (unsigned int*)malloc(array211->size*sizeof(unsigned int));
         memset(array211->data, 0, array211->size*sizeof(unsigned int));
         DEBUG_NEW(array211->id);
      }
      unsigned int loop61 = 0;
      unsigned int loopLimit61 = (rand()%loopsFactor)/2 + 1;
      for(; loop61 < loopLimit61; loop61++) {
         for (int i = 0; i < array211->size; i++) {
            array211->data[i]--;
         }
         for (int i = 0; i < array207->size; i++) {
            if (array207->data[i] == 49) { 
               return array207;
            }
         }
      }
      array_t_param params1;
      params1.size = 3;
      params1.data = (array_t**)malloc(params1.size*sizeof(array_t*));
      params1.data[0] = array207;
      params1.data[1] = array208;
      params1.data[2] = array211;
      array_t* array212 = func24(&params1, get_path(), loopsFactor);
      DEBUG_RETURN(array212->id);
      free(params1.data);
      array212->refC--;
      if(array212->refC == 0) {
         free(array212->data);
         free(array212);
         DEBUG_FREE(array212->id);
      }
      array211->refC--;
      if(array211->refC == 0) {
         free(array211->data);
         free(array211);
         DEBUG_FREE(array211->id);
      }
   }
   else {
      array_t_param params1;
      params1.size = 2;
      params1.data = (array_t**)malloc(params1.size*sizeof(array_t*));
      params1.data[0] = array207;
      params1.data[1] = array208;
      array_t* array213 = func16(&params1, get_path(), loopsFactor);
      DEBUG_RETURN(array213->id);
      free(params1.data);
      array213->refC--;
      if(array213->refC == 0) {
         free(array213->data);
         free(array213);
         DEBUG_FREE(array213->id);
      }
   }
   array207->refC--;
   if(array207->refC == 0) {
      free(array207->data);
      free(array207);
      DEBUG_FREE(array207->id);
   }
   return array208;
}

File: data/multilang_comparison/program_go/src/func9.go
package main
import "math/rand"
var _ = rand.Intn
func Func9(vars *ArrayParam, PATH0 uint64 , loopsFactor int) *Array{
   var pCounter int = vars.Size;
   _ = pCounter
   var array175 *Array
   if pCounter > 0 {
       array175 = vars.Data[pCounter-1]
       pCounter--
       array175.RefC++
   } else {
       array175 = &Array{
           Size: 641,
           RefC: 1,
           Id: 175,
           Data: make([]uint32, 641),
       }
   }
   var params0 ArrayParam
   params0.Size = 1
   params0.Data = make([]*Array, params0.Size)
   params0.Data[0] = array175
   var array176 *Array = Func15(&params0, loopsFactor);
   params0.Data = nil
   params0.Size = 0
   if (PATH0 & 1) != 0 {
      var array179 *Array
      if pCounter > 0 {
          array179 = vars.Data[pCounter-1]
          pCounter--
          array179.RefC++
      } else {
          array179 = &Array{
              Size: 260,
              RefC: 1,
              Id: 179,
              Data: make([]uint32, 260),
          }
      }
      var loop52 int = 0;
      var loopLimit52 int = (50)/2 + 1;
      for ; loop52 < loopLimit52; loop52++ {
         for i := 0; i < array176.Size; i++ {
             array176.Data[i]--
         }
         for i := 0; i < array179.Size; i++ {
             if array179.Data[i] == 5 {
                 return array179
             }
         }
      }
      var params1 ArrayParam
      params1.Size = 3
      params1.Data = make([]*Array, params1.Size)
      params1.Data[0] = array175
      params1.Data[1] = array176
      params1.Data[2] = array179
      var array180 *Array = Func24(&params1, getPath(), loopsFactor);
      params1.Data = nil
      params1.Size = 0
      array180.RefC--
      if array180.RefC == 0 {
          array180.Data = nil
          array180 = nil
      }
      array179.RefC--
      if array179.RefC == 0 {
          array179.Data = nil
          array179 = nil
      }
   
   } else {
      var params1 ArrayParam
      params1.Size = 2
      params1.Data = make([]*Array, params1.Size)
      params1.Data[0] = array175
      params1.Data[1] = array176
      var array181 *Array = Func16(&params1, getPath(), loopsFactor);
      params1.Data = nil
      params1.Size = 0
      array181.RefC--
      if array181.RefC == 0 {
          array181.Data = nil
          array181 = nil
      }
   }
   array175.RefC--
   if array175.RefC == 0 {
       array175.Data = nil
       array175 = nil
   }
   return array176;
}

